airflow/models/dagrun.py

```diff
@@ -1538,11 +1538,21 @@ def schedule_tis(
                 and not ti.task.outlets
             ):
                 dummy_ti_ids.append((ti.task_id, ti.map_index))
-            elif ti.task.start_from_trigger is True and ti.task.start_trigger_args is not None:
-                ti.start_date = timezone.utcnow()
-                if ti.state != TaskInstanceState.UP_FOR_RESCHEDULE:
-                    ti.try_number += 1
-                ti.defer_task(exception=None, session=session)
+            # check whether the operator supports start execution from triggerer
+            elif ti.task.start_trigger_args is not None:
+                from airflow.models.mappedoperator import MappedOperator
+
+                if isinstance(ti.task, MappedOperator):
+                    context = ti.get_template_context()
+                    ti.task._expand_start_trigger(context=context, session=session)
```

Comment:
The thing I’m not confortable here is we’re modifying the MappedOperator in-place. I wonder if we can do something like this instead:

```python
if isinstance(ti.task, MappedOperator):
    start_from_trigger = ti.task.expand_start_from_trigger(context, session)
else:
    start_from_trigger = ti.task.start_from_trigger
if start_from_trigger:
    ...
```

and we can do the rest of the expansion in `defer_task` instead, also in a similar fashion that does not modify the MappedOperator instance, but only the unmapped BaseOperator.

Reply:
I just split `_expand_start_trigger` into `_expand_start_from_trigger` and `_expand_start_trigger_args`. Both of them does not do anything in place and now look a bit better.